addon/mixins/base.js

```diff
@@ -1,138 +1,256 @@
+    ignorableAttrs = ignorableAttrs.concat(EMBER_ATTRS);
+    ignorableAttrs = ignorableAttrs.concat(HTML_ATTRS);
+    ignorableAttrs = ignorableAttrs.concat(CUSTOM_ATTRS);
+    return Ember.A(ignorableAttrs);
+  },
 
-      if (typeof prop === 'function' && typeof (this.get(key) || this.get(`_${key}`)) !== 'function') {
-        continue;
-      }
+  getSemanticScope() {
+    if (Ember.isPresent(this.get('onElement'))) {
+      return this.$(this.get('onElement'));
+    }
+    return this.$();
+  },
 
-      if (window.$.inArray(key, EMBER) >= 0) {
-        value = this.get(`ui_${key}`);
-      } else {
-        if (typeof this.get(key) !== 'undefined') {
-          value = this.get(key);
-        } else {
-          value = this.get(`_${key}`);
-        }
-      }
+  getSemanticModuleName() {
+    return this.get('module');
+  },
 
-      if (value != null) {
-        if (typeof value === 'function') {
-          custom[key] = Ember.run.bind(this, this.updateFunctionWithParameters(key, value));
-        } else {
-          custom[key] = value;
-        }
+  getSemanticModule() {
+    let selector = this.getSemanticScope();
+    if (selector != null) {
+      let module = selector[this.getSemanticModuleName()];
+      if (typeof module === 'function') {
+        return module;
       }
     }
+    return null;
+  },
 
-    return custom;
+  getSemanticModuleGlobal() {
+    let moduleName = this.getSemanticModuleName();
+    return window.$.fn[moduleName];
   },
 
-  updateProperty(property) {
-    return function() {
-      this.execute('set ' + property, this.get(property));
-    };
+  willInitSemantic(settings) { // jshint ignore:line
+    // Use this method to modify the settings object on inherited components, before module initialization
   },
 
-  updateFunctionWithParameters(key, fn) {
-    return function() {
-      var args = [].splice.call(arguments, 0);
-      var internal = this.get(`_${key}`);
+  initSemanticModule() {
+    let module = this.getSemanticModule();
+    if (module) {
+      module.call(this.getSemanticScope(), this._settings());
+    } else {
+      Ember.Logger.error(`The Semantic UI module ${this.getSemanticModuleName()} was not found and did not initialize`);
+    }
+  },
 
-      if (internal) {
-        internal.apply(this, args);
-      }
+  didInitSemantic() {
+    // Use this method after the module is initialized to do post initialized changes
+  },
 
-      if (internal !== fn) {
-        return fn.apply(this, [this].concat(args));
-      }
+  getSemanticAttr(attrName) {
+    return this.execute(`get ${attrName}`);
+  },
 
-      return true;
-    };
+  setSemanticAttr(attrName, attrValue) {
+    return this.execute(`set ${attrName}`, attrValue);
   },
 
-  initializeModule() {
-    this.$()[this.get("module")](this.settings(this.get("module")));
+  areAttrValuesEqual(attrName, attrValue, moduleValue) {
+    return attrValue === moduleValue ||
+           this._stringCompare(attrValue) === moduleValue ||
```

Comment:
This line and the next feel like overkill. What use-case do we have where those would be true but _stringCompare on both sides wouldn't be?